master/pkg/model/experiment.go

```diff
@@ -284,21 +284,23 @@ func NewTrial(
 
 // Step represents a row from the `steps` table.
 type Step struct {
-	TrialID   int        `db:"trial_id"`
-	ID        int        `db:"id"`
-	State     State      `db:"state"`
-	StartTime time.Time  `db:"start_time"`
-	EndTime   *time.Time `db:"end_time"`
-	Metrics   JSONObj    `db:"metrics"`
+	TrialID        int        `db:"trial_id"`
+	ID             int        `db:"id"`
+	State          State      `db:"state"`
+	StartTime      time.Time  `db:"start_time"`
+	EndTime        *time.Time `db:"end_time"`
+	BatchesPerStep int        `db:"batches_per_step"`
```

Comment:
What do you think about renaming this column in the database to `num_steps` or something?  `batches_per_step` made more sense when it was constant.  In most cases of this PR I'm ok with not updating the name yet, especially since in a lot of places `batches_per_step` is going to become `scheduling_unit` or something, but when you are attaching it to a step, it's slightly different.  This value does not refer to the `batches_per_step` config value but rather the concrete value that we are assigning to this step.

Separately, have you thought about the merits and drawbacks of including a `batches_completed` here as well?  I don't know if that's a good idea or not, it's just a thought.

Reply:
drawbacks of including `batches_completed` was just that it wasn't entirely necessary yet, but I can include it. also:

> What do you think about renaming this column in the database to `num_steps` or something? 

I assume you meant `num_batches`. And yeah, since batches_per_step is going to get renamed shortly, I'll hold off on a totally overhaul now.

Reply:
Went ahead and changed it to `num_batches`. Thanks for the suggestion 👍